ocs_ci/utility/utils.py

```diff
@@ -2282,6 +2288,113 @@ def create_ceph_health_cmd(namespace):
     return ceph_health_cmd
 
 
+def run_ceph_health_cmd(namespace):
+    """
+    Run the ceph health command
+
+    Args:
+        namespace: Namespace of OCS
+
+    Raises:
+        CommandFailed: In case the rook-ceph-tools pod failed to reach the Ready state.
+
+    Returns:
+        str: The output of the ceph health command
+
+    """
+    wait_for_rook_ceph_tools_pod(namespace)
+    ceph_health_cmd = create_ceph_health_cmd(namespace)
+    return run_cmd(ceph_health_cmd)
+
+
+def run_faas_consumer_ceph_health_cmd(namespace):
+    """
+    Run the ceph health command on a faas consumer cluster
+
+    Args:
+        namespace: Namespace of OCS
+
+    Raises:
+        CommandFailed: In case the rook-ceph-tools pod failed to reach the Ready state
+
+    Returns:
+        str: The output of the ceph health command
+
+    """
+    try:
+        wait_for_rook_ceph_tools_pod(namespace, timeout=120)
+        ceph_health_cmd = create_ceph_health_cmd(namespace)
+        health = run_cmd(ceph_health_cmd)
+    except CommandFailed as ex:
+        error_msg = "no matching resources found"
+        if error_msg in str(ex) and config.is_provider_exist():
+            health = run_faas_consumer_ceph_health_cmd_on_provider()
+        else:
+            raise ex
+
+    return health
+
+
+def run_faas_consumer_ceph_health_cmd_on_provider():
+    """
+    Run the ceph health command on a faas consumer cluster using the provider rook-ceph-tools pod
+
+    Raises:
+        CommandFailed: In case the rook-ceph-tools pod failed to reach the Ready state
+
+    Returns:
+        str: The output of the ceph health command
+
+    """
+    # Import here to avoid circular loop
+    from ocs_ci.ocs.resources.pod import get_ceph_tools_pod
+
+    log.info("Trying to get the rook-ceph-tools pod of the provider cluster...")
+    try:
+        ct_pod = get_ceph_tools_pod()
+    except (AssertionError, CephToolBoxNotFoundException) as ex:
+        raise CommandFailed(ex)
+
+    return ct_pod.exec_ceph_cmd("ceph health", format=None)
+
+
+def run_ms_consumer_ceph_health_cmd(namespace):
+    """
+    Run the ceph health command on an MS consumer cluster
+
+    Args:
+        namespace: Namespace of OCS
+
+    Raises:
+        CommandFailed: In case the rook-ceph-tools pod failed to reach the Ready state
+
+    Returns:
+        str: The output of the ceph health command
+
+    """
+    # Import here to avoid circular loop
+    from ocs_ci.ocs.managedservice import (
+        patch_consumer_toolbox,
+        is_rados_connect_error_in_ex,
+    )
+
+    wait_for_rook_ceph_tools_pod(namespace, timeout=120)
+    ceph_health_cmd = create_ceph_health_cmd(namespace)
+    try:
+        health = run_cmd(ceph_health_cmd)
+    except CommandFailed as ex:
+        if is_rados_connect_error_in_ex(ex):
+            log.info("Patch the consumer rook-ceph-tools deployment")
+            patch_consumer_toolbox()
```

Comment:
I think that we should directly update the function:
https://github.com/red-hat-storage/ocs-ci/blob/d322ca8ea7d179e229319e6faaa9739432022c63/ocs_ci/ocs/resources/pod.py#L690
Which will do patch on consumer toolbox pod. WDYT?

If we can get info that the command is running on cluster which is consumer I think it can be without any additional parameter, or add one more parameter to be able to enable such patching for consumer cluster for toolbox pod.

Reply:
It should be fine, but I am not 100% sure because the function returns the rook-ceph-tools pod and doesn't say anything about patching. IMO, it will be ok to add the patching as you suggested.

Reply:
In this function we call `setup_ceph_toolbox`

https://github.com/red-hat-storage/ocs-ci/blob/d322ca8ea7d179e229319e6faaa9739432022c63/ocs_ci/ocs/utils.py#L776

You can see we already do something similar for external mode deployment, so I think that the patch logic when we create new ceph_toolbox_pod should be done in this function.